homeassistant/components/mqtt/mixins.py

```diff
@@ -873,7 +878,11 @@ async def discovery_callback(payload: MQTTDiscoveryPayload) -> None:
                 if old_payload != self._discovery_data[ATTR_DISCOVERY_PAYLOAD]:
                     # Non-empty, changed payload: Notify component
                     _LOGGER.info("Updating component: %s", self.entity_id)
-                    await self._discovery_update(payload)
+                    try:
```

Comment:
Here we have a similar case `send_discover_done` should only be send if we throw.